onnxruntime/test/providers/qnn/qnn_test_utils.h

```diff
@@ -30,23 +31,19 @@ struct QuantParams {
   QType zero_point;
 
   static QuantParams<QType> Compute(float rmin, float rmax) {
-    if (rmin == 0.0f && rmax == 0.0f) {  // Quantizing a single zero.
-      return QuantParams<QType>{1.0f, 0};
-    }
+    // Ensure a minimum range of 0.0001 (required by QNN)
+    rmax = std::max(rmax, rmin + 0.0001f);
 
-    if (rmin == rmax) {  // One data-point (x) to quantize.
-      if (rmin < 0) {    // new range is [-x , 0.0f]
-        rmax = 0.0f;
-      } else {  // new range is [0.0f, x]
-        rmin = 0.0f;
-      }
-    }
+    // Both QNN and ORT require the range to include 0.0f
+    rmin = std::min(rmin, 0.0f);
+    rmax = std::max(rmax, 0.0f);
 
     constexpr float qmin = static_cast<float>(std::numeric_limits<QType>::min());
     constexpr float qmax = static_cast<float>(std::numeric_limits<QType>::max());
 
-    const float scale = (rmax - rmin) / (qmax - qmin);
-    const QType zero_point = static_cast<QType>(std::roundf((qmin - rmin) / scale));
+    const float scale = rmax == rmin ? 1.0f : (rmax - rmin) / (qmax - qmin);
+    const float initial_zero_point = qmin - rmin / scale;
+    const QType zero_point = static_cast<QType>(RoundHalfToEven(std::max(qmin, std::min(qmax, initial_zero_point))));
```

Comment:
Now using ORT util for computing the final zero point. Still need to calculate the scale separately due to additional requirements on the floating-point range specified by QNN docs.